### PR TITLE
DebugInfo: fix debugging on SIL level

### PR DESCRIPTION
**Pull request by eeckstein**
The fix is to set the DebugCompilationDir when -gsil is used.
Otherwise no debug info is generated if the source file name does not contain a path.

rdar://problem/49796695

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci smoke test

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -902,21 +902,20 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
...
-      Opts.DebugCompilationDir = cwd.str();
+  }
+  if (Opts.DebugInfoLevel >= IRGenDebugInfoLevel::LineTables) {
+    if (Args.hasArg(options::OPT_debug_info_store_invocation)) {
+      ArgStringList RenderedArgs;
+      for (auto A : Args)
+        A->render(Args, RenderedArgs);
+      CompilerInvocation::buildDebugFlags(Opts.DebugFlags,
+                                          RenderedArgs, SDKPath,
+                                          ResourceDir);
```
Comment (adrian-prantl) [inline lib/Frontend/CompilerInvocation.cpp]: This seems like an unrelated change that happens to not be tested. It doesn't really hurt either.

File: lib/Frontend/CompilerInvocation.cpp [hunk 2]
```diff
@@ -902,21 +902,20 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
...
     }
+    // TODO: Should we support -fdebug-compilation-dir?
+    llvm::SmallString<256> cwd;
+    llvm::sys::fs::current_path(cwd);
+    Opts.DebugCompilationDir = cwd.str();
```
Comment (adrian-prantl) [inline lib/Frontend/CompilerInvocation.cpp]: Looks reasonable!

**Review by adrian-prantl — APPROVED** [review]

**Comment by eeckstein** [comment]
@swift-ci smoke test linux
